rasa/cli/arguments/data.py

```diff
@@ -14,7 +14,6 @@ def set_convert_arguments(parser: argparse.ArgumentParser, data_type: Text):
     parser.add_argument(
         "-f",
         "--format",
-        required=True,
```

Comment:
otherwise, if you do not specify `--format`, it gives this error:
```
usage: rasa data convert nlg [-h] [-v] [-vv] [--quiet] -f {json,md,yaml}
                             --data DATA --out OUT [-l LANGUAGE]
rasa data convert nlg: error: the following arguments are required: -f/--format
```

Reply:
there might be a lot of other occurrences of this

Reply:
I think I'd still keep the `required`! 
We're adding a new sub command `rasa data convert responses`, and it can have it's custom params.

Reply:
but why specify a default argument then? I'm confused

Reply:
to be even clearer: the command's `--help` says:
> Output format the training data should be converted
                        into. Note: currently training data can be converted
                        to 'yaml' format only from 'md' format (**default: yaml**)

So I think we should either:
1. remove `required=True` and keep the default behaviour
2. remove "default: yaml" from the help message because it's misleading

I went with 1. because the UX seemed better to me